crates/apollo-router-core/src/traits.rs

```diff
@@ -55,6 +55,19 @@ where
 
 impl<T: ?Sized> WithCaching for T where T: QueryPlanner + Sized {}
 
+/// An object that accepts a [`Request`] and allow creating [`Route`]'s.
```

Comment:
Be nice to explain in the function doc why `ResponseStream` is returned as the error part of the `Result`, it's GraphQL and all that.

Reply:
Good point! Fixed in ddcefe2829b9841e0d04398613e74f46110f4e4c

I'm a bit in doubt if this should be a ResponseStream or a Response (single response object)... :thinking: It doesn't seem likely that erroring will result in a stream of errors. Or maybe it will? What do you think?

Reply:
I think it's nice as is (ResponseStream), the caller doesn't have to deal with different types depending on if it was an error or not.